terraform/shared/Makefile

```diff
@@ -31,10 +31,14 @@ plan: validate
 rename_resources: config
```

Comment:
`terraform/gitlab/Makefile` is a symlink to this file. We either have to break the link and make a copy or we find a more elegant way to invoke the script as part of `terraform apply`. We should try a `local-exec` provisioner. Now there is no way to really test a local provisioner with actually deleting the resource and recreating. With this in mind I now, again, change my mind and think that we should use the destroy-then-recreate strategy for upgrading because that covers the same code paths as for the initial deployment in an empty account.